daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -89,6 +98,28 @@ object Error {
           case _ => None
         }
     }
+
+    final case class Type(
+        typ: Ast.Type,
+        value: Value[Value.ContractId],
+        override val msg: String,
+    ) extends Error
+
+    final case class ValueNesting(value: Value[Value.ContractId]) extends Error {
+      override def msg: String =
+        s"Provided value exceeds maximum nesting level of ${Value.MAXIMUM_NESTING}"
+    }
+
+    final case class RootNode(nodeId: NodeId, override val msg: String) extends Error
+
+    final case class ContractIdFreshness(
```

Comment:
this _only_ covers preprocessing right? Not the contracd id freshness during interpretation?

Reply:
Yes.